### ISSUE TITLE
`func-style` rule, arrow functions, and `this` keywords.

### ISSUE THREAD

**Issue by mysticatea** [eslint/eslint#3819]
If it's using `this` keyword in an arrow function, the arrow function cannot be replaced with `FunctionDeclaration`.
But currently, the `func-style` rule is warning the arrow function.

**Code:**

```js
/* eslint-env es6 */
/* eslint func-style: [2, "declaration"] */

const foo = function() { this; }.bind(this); // using lexical `this`.
const bar = () => this; // using lexical `this`.
```

**Actual:**

![image](https://cloud.githubusercontent.com/assets/1937871/9900703/73e38c0e-5c9d-11e5-86c2-3b3993d2624c.png)

**Expected:**

No errors.

----

I will work on this if accepted.

**Comment by nzakas** [comment]
Nice catch
